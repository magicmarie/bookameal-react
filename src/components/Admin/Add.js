import React, { Component } from "react";
import { notify } from "react-notify-toast";
import axiosInstance from "../common/Apicalls";

class Add extends Component {
  state = {
    meal_name: "",
    price: "",
    message: ""
  };
  handleNewMeal = event => {
    const { meal_name, price } = this.state;
    // mtd to stop the default action of an element
    event.preventDefault();
    //create new meal
    axiosInstance
      .post("/meals", { meal_name, price })
      .then(response => {
        this.reset();
        //meal created: show success message, close the modal and update th meals list
        notify.show(response.data.message, "success", 2500);
        document.getElementById(`closeAddModal`).click();
        this.props.getMeals();
      })
      // meal not created, show errors
      .catch(error => {
        if (error.response) {
          this.setState({ message: error.response.data.message });
        } else if (error.request) {
          notify.show("Network error", "error", 2500);
        }
      });
  };
  onChange = event =>
    this.setState({ [event.target.name]: event.target.value });

  reset = () =>
    this.setState({
      meal_name: "",
      price: "",
      message: ""
    });

  render() {
    return (
      <div>
        <div
          className="modal fade"
          id="addModal"
          tabIndex="-1"
          role="dialog"
          aria-labelledby="exampleModalLabel"
          aria-hidden="true"
        >
          <div className="modal-dialog" role="document">
            <div className="modal-content">
              <div className="modal-header">
                <h5 className="modal-title" id="exampleModalLabel">
                  Create New meal
                </h5>
                <button
                  onClick={this.reset}
                  type="button"
                  className="close"
                  data-dismiss="modal"
                  aria-label="Close"
                >
                  <span aria-hidden="true">&times;</span>
                </button>
              </div>
              <div className="modal-body">
                {this.state.message && (
                  <div className="alert alert-danger">
                    {this.state.message}
                    <button
                      type="button"
                      className="close"
                      data-dismiss="alert"
                      arial-label="Close"
                    >
                      <span aria-hidden="true"> &times;</span>
                    </button>
                  </div>
                )}
                <form onSubmit={this.handleNewMeal}>
                  <div className="form-group">
                    <label htmlFor="text">Name:</label>
                    <input
                      type="text"
                      name="meal_name"
                      value={this.state.meal_name}
                      onChange={this.onChange}
                      required
                      className="form-control"
                    />
                  </div>
                  <div className="form-group">
                    <label htmlFor="number">Price:</label>
                    <input
                      type="integer"
                      name="price"
                      value={this.state.price}
                      onChange={this.onChange}
                      required
                      className="form-control"
                    />
                  </div>
                  <div>
                    <button
                      type="button"
                      className="btn btn-secondary"
                      data-dismiss="modal"
                      id="closeAddModal"
                      onClick={this.reset}
                    >
                      Close
                    </button>
                    <button type="submit" className="btn btn-primary ml-3">
                      Save
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
export default Add;
