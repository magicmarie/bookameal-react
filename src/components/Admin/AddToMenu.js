import React from "react";
import Loader from "react-loader";

const AddMenuModal = ({ checked, days, handleAddMenu, isMealSet}) => (
  <div>
    <div
      className="modal fade"
      id="addToMenuModal"
      tabIndex="-1"
      role="dialog"
      aria-labelledby="exampleModalLabel"
      aria-hidden="true"
    >
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header">
            <h5 className="modal-title" id="exampleModalLabel">
              Choose days on which to add the meal
            </h5>
            <button
              type="button"
              className="close"
              data-dismiss="modal"
              aria-label="Close"
            >
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div className="modal-body">
            <Loader loaded={isMealSet} />
            {days.map(day => (
              <div key={day} className="form-check form-group">
                <input
                  type="checkbox"
                  name="day"
                  onChange={evt => checked(evt, day)}
                  className="form-check-input"
                />
                <label className="form-check-label" htmlFor="isadmin">
                  {day}
                </label>
              </div>
            ))}

            <button
              type="button"
              className="btn btn-secondary"
              data-dismiss="modal"
              id="closeAddToMenuModal"
            >
              Close
            </button>
            <button
              onClick={handleAddMenu}
              type="button"
              className="btn btn-primary ml-3"
            >
              Save
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default AddMenuModal;
