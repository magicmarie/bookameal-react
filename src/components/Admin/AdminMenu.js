import React, { Component } from "react";
import Loader from "react-loader";
import { notify } from "react-notify-toast";
import axiosInstance from "../common/Apicalls";
import Menu from "./Menu";
import WeekDays from "../common/WeekDays";
import getCurrentDay from "../common/CurrentDay";

/**
 *
 *@param {string} day
 * @class AdminMenu
 * @extends {Component}
 */
class AdminMenu extends Component {
  state = {
    menus: [],
    loaded: false,
    currentMenu: {
      meals: [],
      name: ""
    }
  };

  /**
   *@returns {menus} menus
   * @memberof AdminMenu
   */
  componentDidMount() {
    this.getMenus();
  }

  // get menu: admin side
  /**
   *@returns {null} null
   * @memberof AdminMenu
   */
  getMenus = () => {
    axiosInstance
      .get("/admin-menus")
      .then(response => {
        const today = getCurrentDay();
        const currentMenu = response.data.menus.find(
          menu => today === menu.name
        );
        this.setState({
          menus: response.data.menus,
          loaded: true,
          currentMenu
        });
      })
      .catch(error => {
        if (error.response) {
          const { status } = error.response;
          if (status === 401) {
            localStorage.removeItem("token");
          }
        } else if (error.request) {
          notify.show("Wrong request", "warning", 2500);
        }
      });
  };

  getMenu = day => {
    day = day.charAt(0).toUpperCase() + day.slice(1);
    // manipulate the menus array from the state to get currentMenu
    const { menus } = this.state;
    const currentMenu = menus.find(menu => day === menu.name);
    this.setState({ currentDay: day, currentMenu });
  };
  // delete meal from the menu: close modal, show success message, get the menu
  ConfirmDeleteMeal = id => {
    axiosInstance.delete(`/menu/${id}`).then(response => {
      notify.show(response.data.message, "success", 2500);
      document.getElementById(`cancelModal${id}`).click();
      this.getMenus();
    });
  };

  /**
   * @returns {any} rendered items
   * @memberof AdminMenu
   */
  render() {
    const { currentMenu, loaded } = this.state;
    const menuDetails =
      currentMenu.meals.length === 0 ? (
        <div>No Menu Found</div>
      ) : (
        currentMenu.meals.map(menu => (
          <div className="col-md-3 col-lg-3" key={menu.id}>
            <Menu
              id={menu.id}
              mealName={menu.meal_name}
              price={menu.price}
              key={menu.id}
              ConfirmDeleteMeal={this.ConfirmDeleteMeal}
            />
          </div>
        ))
      );
    return (
      <div className="row">
        <div className="col-md-2">
          <div className="header">Days</div>
          <ul className="list-group">
            <WeekDays getMenu={this.getMenu} />
          </ul>
        </div>
        <div className="vl" />
        <div className="col-md-9 meal-list">
          <h2 className="header text-center">
            {currentMenu.name}
            's Menu List
          </h2>
          <Loader loaded={loaded}>
            <div className="row">{menuDetails}</div>
          </Loader>
        </div>
      </div>
    );
  }
}
export default AdminMenu;
