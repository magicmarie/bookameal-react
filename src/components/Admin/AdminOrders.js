import React, { Component } from "react";
import Loader from "react-loader";
import { notify } from "react-notify-toast";
import axiosInstance from "../common/Apicalls";
import Pagination from "../common/Pagination";

const Order = props => (
  <div className="menu-item">
    <div className="row">
      <div className="col-md-4">{props.mealName}</div>
      <div className="price col-md-2">{props.price}</div>
      <div className="col-md-3 text-center">{props.customerName}</div>
      <div className="col-sm-3 text-center">
        <select>
          <option value="Delivered">Delivered</option>
          <option value="Not delivered">Not delivered</option>
          <option value="In progress">In progress</option>
        </select>
      </div>
    </div>
  </div>
);

class AdminDashboard extends Component {
  state = {
    total: null,
    orders: [],
    loaded: false,
    nextPage: null,
    previousPage: null,
    currentPage: null,
    pages: null,
    totalCount: null
  };
  //get all orders from admin's meals
  getUserOrders = () => {
    const page = this.state.currentPage || 1;
    axiosInstance
      .get("/orders", { params: { page } })
      .then(response => {
        const {
          total,
          currentPage,
          orders,
          nextPage,
          pages,
          perPage,
          previousPage,
          totalCount
        } = response.data.order_items;
        this.setState({
          total,
          currentPage,
          orders,
          nextPage,
          pages,
          perPage,
          previousPage,
          totalCount,
          loaded: true
        });
      })
      .catch(error => {
        if (error.response) {
          const { status } = error.response;
          if (status === 404) {
            this.setState({
              orders: []
            });
          } else if (status === 401) {
            localStorage.removeItem("token");
          }
        } else if (error.request) {
          notify.show("Wrong request", "warning", 2500);
        }
      });
  };
  //pagination
  changePage = selectedPage => {
    this.setState(
      {
        currentPage: selectedPage
      },
      () => this.getUserOrders()
    );
  };

  // always called before the component's first render
  // fetches the orders list
  componentWillMount() {
    this.getUserOrders();
  }
  render() {
    const {
      total,
      orders,
      loaded,
      currentPage,
      pages,
      nextPage,
      previousPage
    } = this.state;

    const ordersDetails =
      orders.length === 0 ? (
        <div>No orders Found</div>
      ) : (
        orders.map(order => (
          <div key={order.id}>
            <Order
              id={order.id}
              mealName={order.meal_name}
              price={order.price}
              customerName={order.customerName}
            />
          </div>
        ))
      );

    return (
      <div className="container-fluid">
        <h2 className="header text-center">Orders</h2>
        <div className="card mb-2" style={{ width: "10rem" }}>
          <div className="card-body">
            <h5 className="card-title">Total amount</h5>
            <p className="card-text">
              {total}
              UGX
            </p>
          </div>
        </div>
        <Loader loaded={loaded}>
          <div>{ordersDetails}</div>
        </Loader>
        <Pagination
          currentPage={currentPage}
          previousPage={previousPage}
          nextPage={nextPage}
          pages={pages}
          changePage={this.changePage}
        />
      </div>
    );
  }
}
export default AdminDashboard;
