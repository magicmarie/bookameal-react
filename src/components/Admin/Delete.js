import React from "react";

// delete meal modal
const DeleteMeal = props => (
  <div>
    <div id={`deleteModal${props.id}`} className="modal fade">
      <div className="modal-dialog modal-confirm">
        <div className="modal-content">
          <div className="modal-header">
            <h4 className="modal-title">Are you sure?</h4>
            <button
              type="button"
              className="close"
              data-dismiss="modal"
              aria-hidden="true"
            >
              &times;
            </button>
          </div>
          <div className="modal-body">
            <p>
              Do you really want to delete <strong>{props.meal_name}</strong>?
              This process cannot be undone.
            </p>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="btn btn-info"
              data-dismiss="modal"
              id={`cancelModal${props.id}`}
            >
              Cancel
            </button>
            <button
              type="button"
              className="btn btn-danger"
              onClick={() => props.ConfirmDelete(props.id)}
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default DeleteMeal;
