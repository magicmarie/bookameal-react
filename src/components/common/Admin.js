// adminRoute: to protect admin routes
import React from "react";
import { Redirect, Route } from "react-router-dom";
import jwtDecode from "jwt-decode";

const AdminRoute = ({ component: Component, ...rest }) => {
  //check the localStorage for the token: if missing, redirect to login.
  // if exists, decode the token and check for is_admin value = True.
  // if True, load the component, else redirect to user
  const token = localStorage.getItem("token");

  return (
    <Route
      // other props being passed down to the route as well
      {...rest}
      render={props => {
        return token === null ? (
          <Redirect to="/login" />
        ) : jwtDecode(token).is_admin === "True" ? (
          <Component {...props} />
        ) : (
          <Redirect to="/user" />
        );
      }}
    />
  );
};

export default AdminRoute;
