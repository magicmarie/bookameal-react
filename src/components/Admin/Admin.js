import React, { Component } from "react";
import Loader from "react-loader";
import { notify } from "react-notify-toast";
import axiosInstance from "../common/Apicalls";
import Meal from "./Meal";
import Add from "./Add";
import AddToMenu from "./AddToMenu";
import Pagination from "../common/Pagination";

class AdminDashboard extends Component {
  state = {
    meals: [],
    loaded: false,
    days: [],
    mealId: 0,
    isMealSet: true,
    nextPage: null,
    previousPage: null,
    currentPage: null,
    pages: null,
    totalCount: null
  };
  //get all meals created by admin
  getMeals = () => {
    // create a variable for page to be current page
    const page = this.state.currentPage || 1;
    axiosInstance
      .get("/meals", { params: { page } })
      .then(response => {
        const {
          currentPage,
          meals,
          nextPage,
          pages,
          perPage,
          previousPage,
          totalCount
        } = response.data.meal_items;
        this.setState({
          currentPage,
          meals,
          nextPage,
          pages,
          perPage,
          previousPage,
          totalCount,
          loaded: true
        });
      })
      .catch(error => {
        if (error.response) {
          const { status } = error.response;
          if (status === 404) {
            this.setState({
              meals: []
            });
          } else if (status === 401) {
            localStorage.removeItem("token");
          }
        } else if (error.request) {
          notify.show("Wrong request", "warning", 2500);
        }
      });
  };

  // always called before the component's first render
  // fetches the meals list
  componentDidMount() {
    this.getMeals();
  }

  //delete meal from meals list: close modal, show success message, get all meals
  ConfirmDelete = id => {
    axiosInstance.delete(`/meals/${id}`).then(response => {
      notify.show(response.data.message, "success", 2500);
      document.getElementById(`cancelModal${id}`).click();
      this.getMeals();
    });
  };

  //edit meal on meals list
  EditMeal = (id, meal_name, price) => {
    axiosInstance
      .put(`/meals/${id}`, {
        meal_name,
        price
      })
      .then(response => {
        this.setState({
          meal_name: "",
          price: "",
          message: ""
        });
        notify.show(response.data.message, "success", 2500);
        document.getElementById(`closeEditModal${id}`).click();
        this.getMeals();
      })
      .catch(error => {
        if (error.response) {
          this.setState({ message: error.response.data.message });
        } else if (error.request) {
          notify.show("Network error", "error", 2500);
        }
      });
  };

  checked = (evt, day) => {
    const { days } = this.state;
    if (evt.target.checked) {
      days.push(day);
    } else {
      days.pop(day);
    }
  };

  setMealId = id => {
    this.setState({ mealId: id });
  };
  //pagination
  changePage = selectedPage => {
    this.setState(
      {
        currentPage: selectedPage
      },
      () => this.getMeals()
    );
  };

  //add meal to menu
  handleAddMenu = () => {
    this.setState({ isMealSet: false });
    const { mealId, days } = this.state;
    days.forEach(day => {
      axiosInstance
        .post(`/menu/${day}/${mealId}`)
        .then(response => {
          notify.show(response.data.message, "success", 2500);
          document.getElementById("closeAddToMenuModal").click();
        })
        .catch(error => {
          if (error.response) {
            const { status } = error.response;
            if (status === 404) {
              this.setState({
                menu: []
              });
            }
            if (status === 409) {
              notify.show(error.response.data.message, "error", 2500);
            }
          } else if (error.request) {
            notify.show("Network error", "error", 2500);
          }
        });
    });
    this.setState({ isMealSet: true });
  };
  render() {
    const {
      meals,
      loaded,
      isMealSet,
      currentPage,
      pages,
      nextPage,
      previousPage
    } = this.state;
    const days = [
      "Monday",
      "Tuesday",
      "Wednesday",
      "Thursday",
      "Friday",
      "Saturday",
      "Sunday"
    ];
    // if no meals, return message. else map thru the meals: meal card
    const mealDetails =
      meals.length === 0 ? (
        <div>No Meals Found</div>
      ) : (
        meals.map(meal => (
          <div className="col-md-3 col-lg-3" key={meal.id}>
            <Meal
              handleAddMenu={this.handleAddMenu}
              ConfirmDelete={() => this.ConfirmDelete(meal.id)}
              EditMeal={this.EditMeal}
              id={meal.id}
              setMealId={this.setMealId}
              mealName={meal.name}
              price={meal.price}
              key={meal.id}
              message={this.state.message}
            />
          </div>
        ))
      );

    return (
      <div className="container-fluid">
        <button
          type="button"
          className="btn btn-primary"
          data-toggle="modal"
          data-target="#addModal"
        >
          New Meal
        </button>
        <hr />
        <div className="row">
          <div className="col meal-list">
            <h2 className="header text-center">Meals List</h2>
            <Loader loaded={loaded}>
              <div className="row">{mealDetails}</div>
            </Loader>
            <Pagination
              currentPage={currentPage}
              previousPage={previousPage}
              nextPage={nextPage}
              pages={pages}
              changePage={this.changePage}
            />
          </div>
        </div>
        <Add getMeals={this.getMeals} />
        <AddToMenu
          handleAddMenu={this.handleAddMenu}
          checked={this.checked}
          days={days}
          isMealSet={isMealSet}
        />
      </div>
    );
  }
}
export default AdminDashboard;
