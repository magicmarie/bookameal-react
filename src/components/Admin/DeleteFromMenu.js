import React from "react";
import { CloseModalButton } from "../common/Helper";

// delete meal from the menu modal
const DeleteMealFromMenu = props => (
  <div>
    <div id={`deleteFromMenuModal${props.id}`} className="modal fade">
      <div className="modal-dialog modal-confirm">
        <div className="modal-content">
          <div className="modal-header">
            <h4 className="modal-title">Are you sure?</h4>
            <CloseModalButton />
          </div>
          <div className="modal-body">
            <p>
              Do you really want to delete <strong>{props.mealName}</strong>{" "}
              from the menu? This process cannot be undone.
            </p>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="btn btn-info"
              data-dismiss="modal"
              id={`cancelModal${props.id}`}
            >
              Cancel
            </button>
            <button
              type="button"
              className="btn btn-danger"
              onClick={() => props.ConfirmDeleteMeal(props.id)}
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default DeleteMealFromMenu;
