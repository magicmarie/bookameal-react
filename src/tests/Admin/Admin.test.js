import React from "react";
import { shallow } from "enzyme";
import Admin from "../../components/Admin/Admin";

describe("AdminDashboard component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Admin />);
  });
  it("renders successfully", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("renders successfully", () => {
    const handleEvent = jest.fn()
    wrapper.find('button.btn-primary').simulate('click', handleEvent)
  });
});
