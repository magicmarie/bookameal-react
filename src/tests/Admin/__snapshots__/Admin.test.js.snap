// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminDashboard component renders successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminDashboard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btn btn-primary"
          data-target="#addModal"
          data-toggle="modal"
          type="button"
>
          New Meal
</button>,
        <hr />,
        <div
          className="row"
>
          <div
                    className="col meal-list"
          >
                    <h2
                              className="header text-center"
                    >
                              Meals List
                    </h2>
                    <Unknown
                              component="div"
                              loaded={false}
                              loadedClassName="loadedContent"
                              parentClassName="loader"
                    >
                              <div
                                        className="row"
                              >
                                        <div>
                                                  No Meals Found
                                        </div>
                              </div>
                    </Unknown>
                    <Pagination
                              changePage={[Function]}
                              currentPage={null}
                              nextPage={null}
                              pages={null}
                              previousPage={null}
                    />
          </div>
</div>,
        <Add
          getMeals={[Function]}
/>,
        <AddMenuModal
          checked={[Function]}
          days={
                    Array [
                              "Monday",
                              "Tuesday",
                              "Wednesday",
                              "Thursday",
                              "Friday",
                              "Saturday",
                              "Sunday",
                            ]
          }
          handleAddMenu={[Function]}
          isMealSet={true}
/>,
      ],
      "className": "container-fluid",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "New Meal",
          "className": "btn btn-primary",
          "data-target": "#addModal",
          "data-toggle": "modal",
          "type": "button",
        },
        "ref": null,
        "rendered": "New Meal",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col meal-list"
>
            <h2
                        className="header text-center"
            >
                        Meals List
            </h2>
            <Unknown
                        component="div"
                        loaded={false}
                        loadedClassName="loadedContent"
                        parentClassName="loader"
            >
                        <div
                                    className="row"
                        >
                                    <div>
                                                No Meals Found
                                    </div>
                        </div>
            </Unknown>
            <Pagination
                        changePage={[Function]}
                        currentPage={null}
                        nextPage={null}
                        pages={null}
                        previousPage={null}
            />
</div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="header text-center"
>
                Meals List
</h2>,
              <Unknown
                component="div"
                loaded={false}
                loadedClassName="loadedContent"
                parentClassName="loader"
>
                <div
                                className="row"
                >
                                <div>
                                                No Meals Found
                                </div>
                </div>
</Unknown>,
              <Pagination
                changePage={[Function]}
                currentPage={null}
                nextPage={null}
                pages={null}
                previousPage={null}
/>,
            ],
            "className": "col meal-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Meals List",
                "className": "header text-center",
              },
              "ref": null,
              "rendered": "Meals List",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="row"
>
                  <div>
                                    No Meals Found
                  </div>
</div>,
                "component": "div",
                "loaded": false,
                "loadedClassName": "loadedContent",
                "parentClassName": "loader",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div>
                    No Meals Found
</div>,
                  "className": "row",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "No Meals Found",
                  },
                  "ref": null,
                  "rendered": "No Meals Found",
                  "type": "div",
                },
                "type": "div",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "changePage": [Function],
                "currentPage": null,
                "nextPage": null,
                "pages": null,
                "previousPage": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "getMeals": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "checked": [Function],
          "days": Array [
            "Monday",
            "Tuesday",
            "Wednesday",
            "Thursday",
            "Friday",
            "Saturday",
            "Sunday",
          ],
          "handleAddMenu": [Function],
          "isMealSet": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btn btn-primary"
            data-target="#addModal"
            data-toggle="modal"
            type="button"
>
            New Meal
</button>,
          <hr />,
          <div
            className="row"
>
            <div
                        className="col meal-list"
            >
                        <h2
                                    className="header text-center"
                        >
                                    Meals List
                        </h2>
                        <Unknown
                                    component="div"
                                    loaded={false}
                                    loadedClassName="loadedContent"
                                    parentClassName="loader"
                        >
                                    <div
                                                className="row"
                                    >
                                                <div>
                                                            No Meals Found
                                                </div>
                                    </div>
                        </Unknown>
                        <Pagination
                                    changePage={[Function]}
                                    currentPage={null}
                                    nextPage={null}
                                    pages={null}
                                    previousPage={null}
                        />
            </div>
</div>,
          <Add
            getMeals={[Function]}
/>,
          <AddMenuModal
            checked={[Function]}
            days={
                        Array [
                                    "Monday",
                                    "Tuesday",
                                    "Wednesday",
                                    "Thursday",
                                    "Friday",
                                    "Saturday",
                                    "Sunday",
                                  ]
            }
            handleAddMenu={[Function]}
            isMealSet={true}
/>,
        ],
        "className": "container-fluid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New Meal",
            "className": "btn btn-primary",
            "data-target": "#addModal",
            "data-toggle": "modal",
            "type": "button",
          },
          "ref": null,
          "rendered": "New Meal",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col meal-list"
>
              <h2
                            className="header text-center"
              >
                            Meals List
              </h2>
              <Unknown
                            component="div"
                            loaded={false}
                            loadedClassName="loadedContent"
                            parentClassName="loader"
              >
                            <div
                                          className="row"
                            >
                                          <div>
                                                        No Meals Found
                                          </div>
                            </div>
              </Unknown>
              <Pagination
                            changePage={[Function]}
                            currentPage={null}
                            nextPage={null}
                            pages={null}
                            previousPage={null}
              />
</div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="header text-center"
>
                  Meals List
</h2>,
                <Unknown
                  component="div"
                  loaded={false}
                  loadedClassName="loadedContent"
                  parentClassName="loader"
>
                  <div
                                    className="row"
                  >
                                    <div>
                                                      No Meals Found
                                    </div>
                  </div>
</Unknown>,
                <Pagination
                  changePage={[Function]}
                  currentPage={null}
                  nextPage={null}
                  pages={null}
                  previousPage={null}
/>,
              ],
              "className": "col meal-list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Meals List",
                  "className": "header text-center",
                },
                "ref": null,
                "rendered": "Meals List",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <div
                    className="row"
>
                    <div>
                                        No Meals Found
                    </div>
</div>,
                  "component": "div",
                  "loaded": false,
                  "loadedClassName": "loadedContent",
                  "parentClassName": "loader",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div>
                      No Meals Found
</div>,
                    "className": "row",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "No Meals Found",
                    },
                    "ref": null,
                    "rendered": "No Meals Found",
                    "type": "div",
                  },
                  "type": "div",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "changePage": [Function],
                  "currentPage": null,
                  "nextPage": null,
                  "pages": null,
                  "previousPage": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "getMeals": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "checked": [Function],
            "days": Array [
              "Monday",
              "Tuesday",
              "Wednesday",
              "Thursday",
              "Friday",
              "Saturday",
              "Sunday",
            ],
            "handleAddMenu": [Function],
            "isMealSet": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
